react/features/recording/components/Recording/StartRecordingDialogContent.js

```diff
@@ -164,18 +284,49 @@ class StartRecordingDialogContent extends Component<Props> {
         );
     }
 
-    _onSwitchChange: boolean => void;
+    _onDropboxSwitchChange: boolean => void;
+    _onRecordingServiceSwitchChange: boolean => void;
 
     /**
      * Handler for onValueChange events from the Switch component.
      *
      * @returns {void}
      */
-    _onSwitchChange() {
+    _onRecordingServiceSwitchChange() {
+
+        // act like group, cannot toggle off
+        if (this.state.jitsiRecordingSelected) {
+            return;
+        }
+
+        this.setState({
+            jitsiRecordingSelected: true,
+            dropboxRecordingSelected: false
+        });
+
         if (this.props.isTokenValid) {
-            this._signOut();
-        } else {
-            this._signIn();
+            this._onSignOut();
+        }
+    }
+
+    /**
+     * Handler for onValueChange events from the Switch component.
+     *
+     * @returns {void}
+     */
+    _onDropboxSwitchChange() {
+        // act like group, cannot toggle off
+        if (this.state.dropboxRecordingSelected) {
+            return;
+        }
+
+        this.setState({
+            jitsiRecordingSelected: false,
+            dropboxRecordingSelected: true
```

Comment:
What about instead setting an enum for the selected service? Would that be tricky or even better at all?

Reply:
I've changed it to using enums.